network/slashing_violations_consumer.go

```diff
@@ -0,0 +1,51 @@
+package network
```

Comment:
please move this file into a separate package `slashing`, then you may rename it as `violations_consumer`.